### PR TITLE
[TIMOB-23542] iOS: Remove all memory leaks and analyzer warnings

### PR DESCRIPTION
**Pull request by cheekiatng**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23542

### CODE REVIEW

File: iphone/Classes/TiAnimation.m [hunk 1]
```diff
@@ -157,8 +157,7 @@ -(void)dealloc
 	RELEASE_TO_NIL(transition);
 	RELEASE_TO_NIL(callback);
 	RELEASE_TO_NIL(view);
-	RELEASE_TO_NIL(animatedView);
```
Comment (hansemannn) [inline iphone/Classes/TiAnimation.m]: Why was this removed? `animatedView` != `animatedViewProxy` :-)

File: iphone/Classes/TiApp.m [hunk 2]
```diff
@@ -1175,12 +1175,14 @@ - (void)dealloc
...
 - (NSString*)systemUserAgent
 {
+#ifndef __clang_analyzer__
     UIDevice *currentDevice = [UIDevice currentDevice];
     NSString *currentLocaleIdentifier = [[NSLocale currentLocale] localeIdentifier];
     NSString *currentDeviceInfo = [NSString stringWithFormat:@"%@/%@; %@; %@;",[currentDevice model],[currentDevice systemVersion],[currentDevice systemName],currentLocaleIdentifier];
     NSString *kTitaniumUserAgentPrefix = [NSString stringWithFormat:@"%s%s%s %s%s","Appc","eler","ator","Tita","nium"];
-    
+    //ignore this on static analyzer
     return [[NSString stringWithFormat:@"%@/%s (%@)",kTitaniumUserAgentPrefix,TI_VERSION_STR,currentDeviceInfo] retain];
```
Comment (hansemannn) [inline iphone/Classes/TiApp.m]: Please describe why we can ignore this

File: iphone/Classes/TiAppiOSUserActivityProxy.m [hunk 3]
```diff
@@ -33,9 +33,10 @@ -(id)initWithOptions:(NSDictionary*)props
 
 -(void)dealloc
 {
-    if(_supported){
+    if (_supported) {
         [self clean];
     }
+	RELEASE_TO_NIL(_userActivity);
```
Comment (hansemannn) [inline iphone/Classes/TiAppiOSUserActivityProxy.m]: Indentation

File: iphone/Classes/TiContactsGroup.m [hunk 4]
```diff
@@ -283,7 +283,10 @@ -(void)add:(id)arg
 					   subreason:nil
 						location:CODELOCATION];
 		};
+		//Ignore static analylzer warning here
```
Comment (hansemannn) [inline iphone/Classes/TiContactsGroup.m]: Same as above, please describe why (because of the release-cycle later, yeah, but only we both know that :-)

Also please try to leave a space in the comments and start capitalized for an improved readability.

**Review by hansemannn — CHANGES_REQUESTED** [review]

File: iphone/Classes/TiAppiOSProxy.m [hunk 5]
```diff
@@ -623,13 +624,14 @@ -(id)scheduleLocalNotification:(id)args
 		CLLocationCoordinate2D center = CLLocationCoordinate2DMake(latitude, longitude);
         
 		if (!CLLocationCoordinate2DIsValid(center)) {
+			RELEASE_TO_NIL(localNotif);
```
Comment (hansemannn) [inline iphone/Classes/TiAppiOSProxy.m]: This line causes the app to crash in my test-case (see JIRA).

**Review by hansemannn — CHANGES_REQUESTED** [review]

File: iphone/Classes/TiApp.m (hunk 2 above)
Comment (cheekiatng) [inline iphone/Classes/TiApp.m]: ha. Sorry we didn't have to retain. I presumed you retained it on purpose in 
https://jira.appcelerator.org/browse/TIMOB-23623
but i just verified that without the retain, the sample app still builds correctly. changes made.

File: iphone/Classes/TiContactsGroup.m (hunk 4 above)
Comment (cheekiatng) [inline iphone/Classes/TiContactsGroup.m]: ok

**Review by hansemannn — APPROVED** [review]

**Comment by hansemannn** [comment]
CR + FT **passed**, PR **approved**!
